x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -93,7 +106,20 @@ private static Failure fail(Node<?> source, String message, Object... args) {
         return new Failure(source, format(Locale.ROOT, message, args));
     }
 
-    static Collection<Failure> verify(LogicalPlan plan) {
+    public Map<Node<?>, String> verifyFailures(LogicalPlan plan) {
+        Collection<Failure> failures = verify(plan);
+        return failures.stream().collect(toMap(Failure::source, Failure::message));
+    }
+    
+    public void metrics(Metrics metrics) {
```

Comment:
Move metrics into the `PlanExecutor` and pass it at construction time to the `Verifier`. There's no need for setters.